### PR TITLE
multi command keybinds

### PR DESCRIPTION
**Pull request by taiyu-len**
list of commands can now binded to a key by `; ,`
for example `bindsym $mod+Return exec urxvt; exec firefox, exec whatever`

commands return a bit more information, which `config_read` handles better.
for example `mode this {` and `}`

### CODE REVIEW

File: sway/commands.c [hunk 1]
```diff
@@ -145,14 +184,19 @@ static enum cmd_status cmd_bindsym(int argc, char **argv) {
...
+	// Put argument into cmd array
+	char *tmp = join_args(argv, argc);
+	char cmd[4096];
+	strcpy(cmd, tmp);
+	free(tmp);
+	
+	char *args[] = {"sh", "-c", cmd, 0 };
 
-	pid_t pid = fork();
+	pid_t pid = vfork();
```
Comment (SirCmpwn) [inline sway/commands.c]: Woah, why did you change to vfork?

File: sway/commands.c [hunk 2]
```diff
@@ -18,6 +18,45 @@
...
+static sway_cmd cmd_set;
+static sway_cmd cmd_split;
+static sway_cmd cmd_splith;
+static sway_cmd cmd_splitv;
+static sway_cmd cmd_workspace;
+static sway_cmd cmd_ws_auto_back_and_forth;
+
+#define NO_BIND() if (!config->reading) return CMD_FAILURE;
+#define NO_CONF() if (config->reading) return CMD_FAILURE;
+#define DEFER() if (!config->active) return CMD_DEFER;
```
Comment (SirCmpwn) [inline sway/commands.c]: No macros.

File: sway/commands.c [hunk 3]
```diff
@@ -960,6 +1007,11 @@ static enum cmd_status cmd_log_colors(int argc, char **argv) {
 	return CMD_SUCCESS;
 }
 
+__attribute__((unused))
+enum cmd_status cmd_for_window(int argc, char **argv) {
+	return CMD_FAILURE;
+}
```
Comment (SirCmpwn) [inline sway/commands.c]: If you aren't using it, take it out.

**Comment by SirCmpwn** [comment]
Looks alright. Need to hear test results from xorg and native before I'll merge.

File: sway/commands.c (hunk 1 above)
Comment (taiyu-len) [inline sway/commands.c]: prevents multiple child process from trying to write to the log at the same time.
for when keybinding multiple cmd_exec.

Comment (SirCmpwn) [inline sway/commands.c]: Well, alright then.

Comment (taiyu-len) [inline sway/commands.c]: well i guess rearranging the sway_log before the fork would also fix it too,
though since all we do is fork/exec thats what vfork was made to do so either way is fine.

**Comment by SirCmpwn** [comment]
bump. Any testers?

**Comment by SirCmpwn** [comment]
w/e
